src/main/scala/com/gu/inspectorlambda/chiefinspector/ChiefInspector.scala

```diff
@@ -44,7 +44,10 @@ object ChiefInspector extends StrictLogging {
     val assessmentTemplates = matchingInstanceSets.map { case (tagCombo, _) =>
       val name = constructName(tagCombo)
       val resourceGroupArn: String = inspector.getResourceGroup(name) getOrElse inspector.createResourceGroup(name)
-      val assessmentTargetArn = inspector.getAssessmentTarget(name, resourceGroupArn) getOrElse inspector.createAssessmentTarget(name, resourceGroupArn)
+      val assessmentTargetArn = inspector.getAssessmentTarget(name, resourceGroupArn) getOrElse {
+        Thread.sleep(5000) // allow deletion of old asessment targets to take effect
```

Comment:
We shouldn't wait here. Maybe between lines 41 and 42, but not here.

We should also check that the target has actually been deleted; otherwise we should wait again.

Reply:
Yeah, looks like this will always wait, but we actually only need to wait if we did the delete?

I agree if it's possible to check the deletion worked a `retryUntil` function based on Futures would be a better solution. This app is all synchronous and `Thread.sleep`y at the moment, so the style is consistent but if we need to be aware of "time" with any business logic (e.g. a retry) then we should do that properly.

Reply:
LL41/42 wouldn't be correct - that's code to cope with deleting stale information left from an old version.  This is the correct place except that it will always sleep before creating a new value, even if an old value did not exist.